### ISSUE TITLE
[Planting Flow] feature to be rethought - make the feature more clear

### ISSUE THREAD

**Issue by ri0nardo** [waterthetrees/wtt_front#314]
When I went to plant a tree, I got confused when the tree green circle icon popped up. I hovered over it and it told me to drag it to the location I wanted to plant a tree. I initially thought the screen was loading, and just sat there for sometimes until I realized nothing happened. Could have text below to guide the user or change the process. I don't think its a hard fix, its just reconsidering how to not make a person confused at the start. I just think it lacks context for anyone.

Things to consider
- Change the icon to not look like a loading bubble.
- Have text to guide the user.
- Have info before clicking plant.
- Make it simple as it is, but adding more context.
- How do mobile users react since this could be a desktop issue. Could be more confusing if the icon is as big it is.
- Reducing the icon size, so it isn't seen as a loading icon.

![Image](https://user-images.githubusercontent.com/98299952/196023190-19015295-53d7-4594-a1c2-20f248fdc519.PNG)

**Comment by ri0nardo** [comment]
@zoobot I wanted to make an IA (Information Architecture) wireframe before I proceed. Still in the process.

**Comment by ri0nardo** [comment]
The icon to land on a location is somewhat more clear on these platforms. While not great, would be ideal to have text explaining what to do above or below. As in having a box that pops up to alert the user what to do helps give direction.

![Image](https://user-images.githubusercontent.com/98299952/204206197-50575581-ec07-47f7-8edb-17719a028606.PNG)



![Image](https://user-images.githubusercontent.com/98299952/204206283-0872177b-b1df-4a21-862b-ba05003ab4ac.PNG)

**Comment by mwpark2014** [comment]
From what @ri0nardo and I talked about, I think we were in agreement that we want to 
1) Change the icon to not look like a loading bubble, maybe something like a pin
2) Allow user to place the icon by clicking on the map
3) Potentially turn off other tree markers when planting new tree

I had a concern about auto-enabling geolocation when activating planting mode on mobile. What if user wants to add in data about an already planted tree that is away from their current location?
4) Maybe mimic uber mobile app in asking user if they would like to be directed to their current location

**Comment by ri0nardo** [comment]
To add an example for the annotations, I was considering using a similar format to uber where theres a bottom section to guide the user. 
![All-Screens](https://user-images.githubusercontent.com/98299952/213375397-f6af78ab-cde0-440c-b925-ea36f01b6bf0.png)

For design purposes, I would say the priority would be
1. Adding indicators of what to do to guide the user.
2. Remove the vector dots from the map when planting.
3. Redesign the icon / graphic where you want want / have planted a tree.
4. TBD

**Comment by ri0nardo** [comment]
@mwpark2014 lets document all the issues onto this ticket with the planting. Plus add all features that you would want to see or consider. Once that is done, we can make a roadmap regarding this issue, so theres direction on what actions need to be done. From there design and development can occur simultaneously plus were working on the same goal. We can break the whole task down to smaller task while we work towards the end goal.

I want to add some annotations on some screens of what is broken, buggy, or wacky.

If you want to document this on google docs we can do it that way if you are more comfortable documenting on there. I normally just document on github since we can edit here, and don't have to open another tab.

**Comment by zoobot** [comment]
Have you played with planting in the field(on mobile) and on desktop? Would be great to have context of what is broken, buggy, or wacky!!

I'd like to keep planting as simple as possible. 

Would this fix the issues?
Add a crosshair for a nose to clarify that it's not a loading bubble? 
Have tooltip text when the icon shows up instead of just on hover?

This bit of code is not super straight forward so expect a chunk of time to deal with this. 
If it is broken, can we fix it with refactoring for code clarity? 
This has been refactored multiple times in the last few years and every time it adds a lot of time intensive weird bugs.

3. Remove the vector dots from the map when planting. <-- What issue is this trying to solve?

**Comment by zoobot** [comment]
It's nice to have everything in git so it's easier to keep track of.

**Comment by ri0nardo** [comment]
@zoobot to answer "Remove the vector dots from the map when planting. <-- What issue is this trying to solve?"

More of a suggestion when planting, there are so many vectors on the screen that its overwhelming, at least to me, the amount of dots, and locate where you are planting a tree. Plus the current planting icon is hard to see with the vectors. I mentioned to @mwpark2014 that it may be easier to see where you are with the vectors turned off when planting. This can be a feature on user settings if they want it on or off when planting. When you click on the legend under "no data" it hides all the trees which is what I was suggesting only when you plant.

Just something to think about. As a user I thought others could relate to the number of dots on screen especially on mobile. I'm adding a screenshot below.
![Capture](https://user-images.githubusercontent.com/98299952/213568166-06d5ecbb-d113-4ade-a7af-b38daac733b8.PNG)

**Comment by zoobot** [comment]
Would it help if their diameter of the green dots became pinpoint or if they were way more transparent? @ri0nardo 
If they completely disappear I'm worried that would seem buggy and also wouldn't be able to see your own tree in context of other trees around you.

**Comment by ri0nardo** [comment]
@zoobot  Theres a few routes we could depending on how you want to approach planting.

1. we can try to make what exist work.
2. we can go for a redesign
3. we can do a mix of both (i choose this)

The first reason the plant icon doesn't necessary work is that the icon looks like a loading symbol. @mwpark2014 and i first experience using the planting feature turned out to be the same, and i guarantee more users have also had a similar experience.

Second the focus of planting isn't too clear once you hit the button. there are so many entities on the screen which confuses users from the start. As an example, if we look at uber driver app (which isn't the same, but has good design examples) the focus is where the user is at. (you can see the image i provided above on a different comment)

I would say there is a valid point of trying to orient yourself to other landmarks, and thats the reason why i mentioned we can have a setting to turn off or on the existing trees on the map.

The main reason i would consider turning off the existing tree vectors is that
1. once the planting pin appears its easy to identify what is the planting pin.
2. It gives the planting pin a focus
3. Theres just too much going on currently.

**Comment by ri0nardo** [comment]
Planning for revisions for planting feature
1. Add Annotations for each action that
2. Find more examples of map pins
3. figure out flow with additions of new features (mass planting, maintenance)
4. How to integrate pictures, uploads within the flow
